src/classes/RLLP_OppRollup.cls

```diff
@@ -80,109 +91,199 @@ public without sharing class RLLP_OppRollup {
+            ' WHERE Opportunity.Id IN :oppIds and IsPrimary = true ALL ROWS ';
+        Map<Id, Contact> contactById = new Map<Id, Contact>();
+        Map<Id, npo02__Household__c> householdById = new Map<Id, npo02__Household__c>();
+
+
+        for (OpportunityContactRole role : Database.query(query)){
+            contactById.put(role.ContactId, role.Contact);
+
+            if (role.Contact.npo02__Household__c != null) {
+                householdById.put(role.Contact.npo02__Household__c, role.Contact.npo02__Household__r);
             }
-        }       
-            
-        if (cmap.size() > 0){              
-            RLLP_OppRollup oppR = new RLLP_OppRollup();
-            oppR.rollupContacts(cmap);   
-            if (hhmap.size() > 0){
-                oppR.rollupHouseholds(hhmap);    
-            }        
-        }           
+
+        }
+
+        if (contactById.isEmpty()) {
+            return;
+        }
+
+        RLLP_OppRollup oppRollup = new RLLP_OppRollup();
+        oppRollup.rollupContacts(contactById);
+
+        if (!householdById.isEmpty()) {
+            oppRollup.rollupHouseholds(householdById);
+        }
+    }
+
+    /**
+     * @description Calls the appropriate rollup engine (legacy or crlp) for the account ids provided
+     * @param accountIds Ids for the accounts that require rollups to be recalculated
+     */
+    public void rollupAccounts(Set<Id> accountIds) {
+        if (accountIds == null || accountIds.isEmpty()) {
+            return;
+        }
+
+        if (isCustomizableRollupsEnabled) {
+            runCustomizableRollups(accountIds);
+
+            return;
+        }
+
+        if (isImmediate(accountIds)) {
+            rollupAccounts(getAccountById(accountIds));
+
+        } else {
+            RLLP_OppRollup.rollupAccountsFuture(accountIds);
+        }
+    }
+
+    /**
+     * @description Calls the appropriate rollup engine (legacy or crlp) for the contact ids provided
+     * @param contactIds Ids for the contacts that require rollups to be recalculated
+     */
+    public void rollupContacts(Set<Id> contactIds) {
+        if (contactIds == null || contactIds.isEmpty()) {
+            return;
+        }
+
+        if (isCustomizableRollupsEnabled) {
+            runCustomizableRollups(contactIds);
+
+            return;
+        }
+
+        if (isImmediate(contactIds)) {
+            rollupContacts(getContactById(contactIds));
+
+        } else {
+            RLLP_OppRollup.rollupContactsFuture(contactIds);
+        }
+    }
+
+    /**
+     * @description Calls the future or immediate customizable rollups depending on current context
+     * @param  ids The ids of the sObjects that require rollups to be recalculated
+     */
+    private void runCustomizableRollups(Set<Id> ids){
+        if (isImmediate(ids)) {
+            CRLP_RollupQueueable.runRollupsForIds(new List<Id>(ids));
+
+        } else {
+            System.enqueueJob(new CRLP_RollupQueueable(new List<Id>(ids)));
```

Comment:
Is there a possibility that this method will be called twice when `ids.size() > 1`? If so, that will throw an exception when `enqueueJob` is called the second time. Reading through the code I don't believe this is possible, but I just want to make sure.

Reply:
The only possibility would be once for account and then once for contact but I'm assuming it'd only result in an error if it was the same object type?

Reply:
It's not the object that would be an issue, but attempting to call `System.enqueueJob()` twice that would throw an exception.

Reply:
@force2b - I added a unit test for this scenario and I ran a test in execute anonymous after loading test data, I am unable to get an exception:
`Set<Id> accountIds = new map<id, account>([select id from account]).keyset();
Set<Id> contactIds = new map<id, contact>([select id from contact]).keyset();

RLLP_OppRollup oppRollup = new RLLP_OppRollup();

oppRollup.rollupAccounts(accountIds);
oppRollup.rollupContacts(contactIds);`

Reply:
You're right. There definitely _was_ a limit at some point, but according to the docs: `You can add up to 50 jobs to the queue with System.enqueueJob in a single transaction.`